redhat-developer/service-binding-operator
# [acceptance-tests] Make acceptance tests compatible with plain k8s clusters

* `TEST_ACCEPTANCE_CLI` env var controls what CLI is used for comunication with cluster. Default: `oc`
* Expose application endpoints via services and node ports instead of Openshift routes
* Tested against minikube with OLM installed
* Tests requiring 3rd party operators are annotated with `@olm`
* Added additional job to Travis CI conf, enabling acceptanced test running against minikube on each PR

### Testing
1. Set `KUBECONFIG` for both minikube and acceptance tests (it will be generated at minikube's start if it does not exist):
```sh
$ export KUBECONFIG=/tmp/minikubeconfig
```
2. Start minikube (v1.14.2):
```sh
$ minikube start
minikube v1.14.2 on Generic 31
    ▪ KUBECONFIG=/tmp/minikubeconfig
...
🏄  Done! kubectl is now configured to use "minikube" by default
```
3. Enable olm on minikube:
```sh
$ minikube addons enable olm
🌟  The 'olm' addon is enabled
```
4. Execute all acceptance tests expect those tagged with `@knative` using `kubectl` CLI:
```sh
$ TEST_ACCEPTANCE_CLI=kubectl TEST_ACCEPTANCE_TAGS="~@knative,~@examples" make test-acceptance
```
